dot/sync/bootstrap_syncer.go

```diff
@@ -0,0 +1,78 @@
+package sync
+
+import (
+	"math/big"
+
+	"github.com/ChainSafe/gossamer/dot/network"
+	"github.com/ChainSafe/gossamer/lib/common"
+)
+
+var _ workHandler = &bootstrapSyncer{}
+
+var bootstrapRequestData = network.RequestedDataHeader + network.RequestedDataBody + network.RequestedDataJustification
+
+// bootstrapSyncer handles worker logic for bootstrap mode
+type bootstrapSyncer struct {
+	blockState BlockState
+}
+
+func newBootstrapSyncer(blockState BlockState) *bootstrapSyncer {
+	return &bootstrapSyncer{
+		blockState: blockState,
+	}
+}
+
+func (s *bootstrapSyncer) handleWork(ps *peerState) (*worker, error) {
+	head, err := s.blockState.BestBlockHeader()
+	if err != nil {
+		return nil, err
+	}
+
+	if ps.number.Cmp(head.Number) <= 0 {
+		return nil, nil
+	}
+
+	return &worker{
+		startHash:    common.EmptyHash,
+		startNumber:  big.NewInt(0).Add(head.Number, big.NewInt(1)),
+		targetHash:   ps.hash,
+		targetNumber: ps.number,
+		requestData:  bootstrapRequestData,
+		direction:    network.Ascending,
+	}, nil
+}
+
+func (s *bootstrapSyncer) handleWorkerResult(res *worker) (*worker, error) {
```

Comment:
it can't receive a `peerState` as a parameter, as this is called when a worker finishes, not when there is a new peer state (and the worker doesn't know what peer state triggered it). it needs the worker so it can handle the result of the worker (success or failure) maybe see the `chain_sync.go` `sync()` function for clarity?

Reply:
ok, is it possible to, instead of receive a `*peerState`, create one, fill with the infos inside the `res`  and pass to the `handleWork` method?

Reply:
not really, since the peerState doesn't contain all hte info that the `res *worker` contains. I think it's cleaner this way, if there was another function that took `*peerState` and `*worker` there woudl have to be a bunch of if statement to determine the worker fields, which would be more confusing imo